rust/ql/lib/codeql/rust/elements/internal/PathPatImpl.qll

```diff
@@ -21,5 +21,9 @@ module Impl {
    * }
    * ```
    */
-  class PathPat extends Generated::PathPat { }
+  class PathPat extends Generated::PathPat {
+    override string toString() { result = this.toAbbreviatedString() }
+
+    override string toAbbreviatedString() { result = this.getPath().toString() }
```

Comment:
Recursion.

Reply:
Hmm, this one (as well as some others) doesn't seem really recursive, as won't be reentrant here (and VSCode doesn't indeed mark it as recursive).

Or would you prefer to avoid it calling into a super-class version of itself ever? Adding something like `getShortName` to `Path` seems like boilerplate (after all, that's what `toAbbreviatedString` is meant to return)

Reply:
`toString` calls `toAbbreviatedString`, so I would expect this call to be recursive.

Reply:
yes, but `PathPat.toAbbreviatedString` calls `Path.toString` (in my local changes I'm changing it to `Path.toAbbreviatedString`), and that doesn't ever go back into `PathPat` (`Path` never contains a `Pat`). Would that still be problematic?

Reply:
The optimizer may or may not be able to infer that there is no recursion. The best thing to do is look at the DIL.

Reply:
Hmm, I don't know how to do that (yet) :sweat_smile:

Reply:
I'm no expert in DIL reading, but I think indeed there is now no recursion involved